checker/src/main/java/org/checkerframework/checker/mustcall/MustCallTransfer.java

```diff
@@ -0,0 +1,351 @@
+   * @param supplier a type factory that can supply the executable elements for CreatesObligation
+   *     and CreatesObligation.List's value elements. Usually, you should just pass atypeFactory
+   *     again. The arguments are different so that the given type factory's adherence to both
+   *     protocols are checked by the type system.
+   * @return the java expression representing the target, or null if the target is unparseable
+   */
+  private static @Nullable JavaExpression getCreatesObligationExpression(
+      AnnotationMirror createsObligation,
+      MethodInvocationNode n,
+      GenericAnnotatedTypeFactory<?, ?, ?, ?> atypeFactory,
+      CreatesObligationElementSupplier supplier) {
+    // Unfortunately, there is no way to avoid passing "this" here. The default must be hard-coded
+    // into the client, such as here. That is the price for the efficiency of not having to query
+    // the annotation definition (such queries are expensive).
+    String targetStrWithoutAdaptation =
+        AnnotationUtils.getElementValue(
+            createsObligation, supplier.getCreatesObligationValueElement(), String.class, "this");
+    // TODO: find a way to also check if the target is a known tempvar, and if so return that. That
+    // should improve the quality of the error messages we give.
+    JavaExpression targetExpr;
+    try {
+      targetExpr =
+          StringToJavaExpression.atMethodInvocation(
+              targetStrWithoutAdaptation, n, atypeFactory.getChecker());
+      if (targetExpr instanceof Unknown) {
+        issueUnparseableError(n, atypeFactory, targetStrWithoutAdaptation);
+        return null;
+      }
+    } catch (JavaExpressionParseException e) {
+      issueUnparseableError(n, atypeFactory, targetStrWithoutAdaptation);
+      return null;
+    }
+    return targetExpr;
+  }
+
+  /**
+   * Issues a mustcall.not.parseable error.
+   *
+   * @param n the node
+   * @param atypeFactory the type factory to use to issue the error
+   * @param targetStrWithoutAdaptation the unparseable string
+   */
+  private static void issueUnparseableError(
+      MethodInvocationNode n,
+      GenericAnnotatedTypeFactory<?, ?, ?, ?> atypeFactory,
+      String targetStrWithoutAdaptation) {
+    atypeFactory
+        .getChecker()
+        .reportError(
+            n.getTree(),
+            "mustcall.not.parseable",
+            n.getTarget().getMethod().getSimpleName(),
+            targetStrWithoutAdaptation);
+  }
+
+  /**
+   * This method either creates or looks up the temp var t for node, and then updates the store to
+   * give t the same type as node.
+   *
+   * @param node the node to be assigned to a temporal variable
+   * @param result the transfer result containing the store to be modified
+   */
+  public void updateStoreWithTempVar(TransferResult<CFValue, CFStore> result, Node node) {
+    // Must-call obligations on primitives are not supported.
+    if (!TypesUtils.isPrimitiveOrBoxed(node.getType())) {
```

Comment:
Should this test for String too?
What is the reason to include boxed primitives?

Reply:
I changed how the checker handles strings entirely, so this comment is no longer relevant.

Reply:
I still have the question about why boxed primitives are forbidden to have a `@MustCall` annotation.

Reply:
Boxed primitives can't have must-call types because primitives can't have must-call types, and the two are convertible. This choice was made basically as a convenience: we didn't want to deal with (boxed or unboxed) primitives accidentally getting must-call types, so we just forbid it in general for both.

I can add a note to the manual if you think it's necessary, but to me it seems obvious that primitives can't have meaningful must-call types, because they're not objects.